pkg/datapath/loader/base.go

```diff
@@ -499,3 +500,50 @@ func (l *Loader) Reinitialize(ctx context.Context, o datapath.BaseProgramOwner,
 
 	return nil
 }
+
+// OnUpdateCiliumNode is only called if running with IPAM ENI and IPsec. In
+// that case, we need to rerun reinitializeIPSec whenever new ENIs are added to
+// the local node, to attach our BPF program for IPsec decryption to the new
+// interfaces.
+func (l *Loader) OnUpdateCiliumNode(oldNode, newNode *ciliumv2.CiliumNode, _ *lock.StoppableWaitGroup) error {
+	if !k8s.IsLocalCiliumNode(newNode) {
+		return nil
+	}
+
+	log.Debug("Checking for new ENIs on local node")
+
+	newENIs := make([]string, 0)
+	for newENI := range newNode.Status.ENI.ENIs {
+		existed := false
+		for oldENI := range oldNode.Status.ENI.ENIs {
+			if newENI == oldENI {
+				existed = true
+				break
+			}
+		}
+		if !existed {
+			newENIs = append(newENIs, newENI)
```

Comment:
What interfaces do we actually want to attach the IPSec program to?

In IPAM, we actually do allow the user to exclude certain interfaces for pod IP allocation. This happens if the user wants to use those interfaces for other purposes than pod IPs, e.g. for egressgw or if they are running non-pod workloads on the node (see `spec.eni.exclude-interface-tags` and `spec.eni.first-interface-index` in https://docs.cilium.io/en/v1.13/network/concepts/ipam/eni/#ipam-eni)

Do we need to have the same logic here, or can the IPSec program be attached to any interface even if it doesn't have anything to do with Cilium?

Checking if an interface is excluded is luckily very easy:
https://github.com/cilium/cilium/blob/3e805306fc4125fca3398397acdf11a13f1a9600/pkg/ipam/crd_eni.go#L50-L52

Reply:
We can attach to any interface currently. That might change once we support IPsec+KPR, but right now, it's fine. We already attach to e.g. the `nodelocaldns` interface :shrug:  Unless those interface receive ESP traffic, the BPF program won't actually do anything.